skillmap/src/components/UserProfile.tsx

```diff
@@ -0,0 +1,130 @@
+import * as React from "react";
+import { connect } from "react-redux";
+import { SkillMapState } from "../store/reducer";
+
+import { Modal } from './Modal';
+
+import { dispatchCloseUserProfile, dispatchShowDeleteAccountModal } from "../actions/dispatch"
+
+interface UserProfileProps {
+    signedIn: boolean;
+    profile: pxt.auth.UserProfile
+    showProfile: boolean;
+    dispatchCloseUserProfile: () => void;
+    dispatchShowDeleteAccountModal: () => void;
+}
+
+
+export class UserProfilImpl extends React.Component<UserProfileProps, {}> {
+    render() {
+        const { showProfile } = this.props;
+
+        if (showProfile) {
+            return this.renderUserProfile();
+        }
+
+        return <div/>
+    }
+
+    renderUserProfile = () => {
+        const user = this.props.profile;
+
+        return <Modal title={user?.idp?.displayName || ""} fullscreen={true} onClose={this.handleOnClose}>
+        <div className="profiledialog">
+            {this.getAccountPanel()}
+            {this.getFeedbackPanel()}
+        </div>
+        </Modal>
+    }
+
+    getAccountPanel = () => {
+        const { profile } = this.props;
+        const provider = profile?.idp?.provider && pxt.auth.identityProvider(profile?.idp?.provider);
+
+        const avatarElem = (
+            <div className="profile-pic avatar">
+                <img src={profile?.idp?.picture?.dataUrl} alt={lf("User")} />
+            </div>
+        );
+        const initialsElem = (
+            <div className="profile-pic avatar">
+                <span>{pxt.auth.userInitials(profile)}</span>
+            </div>
+        );
+
+        return <div className="account panel ui card">
+            <div className="header-text">
+                <label>{lf("Profile")}</label>
+            </div>
+            {profile?.idp?.picture?.dataUrl ? avatarElem : initialsElem}
+            <div className="row-span-two">
+                <label className="title">{lf("Name")}</label>
+                <p className="value">{profile?.idp?.displayName}</p>
```

Comment:
is it possible that the displayName/username/provider.name values will be undefined? (if not then disregard!) otherwise should we have some kind of check?

Reply:
It is possible for displayName to be undefined. Username and provider should always be populated. I think a lot of this was just lifted from webapp.

Reply:
oh yeah I just copied. I'll give this and webapp a pass for undefined ty for noticing!!